src/Native/Unix/System.Native/pal_io.c

```diff
@@ -1001,7 +1001,11 @@ int32_t SystemNative_Read(intptr_t fd, void* buffer, int32_t bufferSize)
     }
 
     ssize_t count;
+#if COREFX_PAL_IN_MONO
+    count = read(ToFileDescriptor(fd), buffer, (uint32_t)bufferSize);
```

Comment:
Same as https://github.com/mono/corefx/pull/80/files#diff-9b1ef5b275691ff247dbf32f834de32eR24, we should not rely on the `System.Native` implementation since we want a very different behavior.